File: TourneyTrack/interface/src/main/java/org/tourneytrack/intf/dto/data/SubmissionDto.java
package org.tourneytrack.intf.dto.data;

import lombok.Data;

import java.util.Date;

@Data
public class SubmissionDto {
    private Long id;
    private Long competitionId;
    private UserDto user;
    private RuleDto rule;
    private Date submittedAt;
    private String description;
    private String status;
    private String reviewerComment;
}

File: TourneyTrackFrontend/tourneytrack/src/app/shared/models/submission.dto.ts
import { UserDto } from './user.dto';
import { RuleDto } from './rule.dto';

export interface SubmissionDto {
  id: number;
  competitionId: number;
  user: UserDto;
  rule: RuleDto;
  submittedAt: string;   // ISO dátum string
  description: string;
  status: string;
  reviewerComment: string;
}
